controllers/embedder_controller.go

```diff
@@ -157,14 +157,26 @@ func (r *EmbedderReconciler) UpdateStatus(ctx context.Context, instance *arcadia
 		if !ok {
 			msg = _StatusNilResponse
 		}
-		// Set status to available
-		newCondition = arcadiav1alpha1.Condition{
-			Type:               arcadiav1alpha1.TypeReady,
-			Status:             corev1.ConditionTrue,
-			Reason:             arcadiav1alpha1.ReasonAvailable,
-			Message:            msg,
-			LastTransitionTime: metav1.Now(),
-			LastSuccessfulTime: metav1.Now(),
+		if instanceCopy.Status.GetCondition(arcadiav1alpha1.TypeReady).Status != corev1.ConditionTrue {
+			// Set status to available, record transition time
+			newCondition = arcadiav1alpha1.Condition{
+				Type:               arcadiav1alpha1.TypeReady,
+				Status:             corev1.ConditionTrue,
+				Reason:             arcadiav1alpha1.ReasonAvailable,
+				Message:            msg,
+				LastTransitionTime: metav1.Now(),
+				LastSuccessfulTime: metav1.Now(),
+			}
+		} else {
+			// Refresh successful time only
+			newCondition = arcadiav1alpha1.Condition{
+				Type:               arcadiav1alpha1.TypeReady,
+				Status:             corev1.ConditionTrue,
+				Reason:             arcadiav1alpha1.ReasonAvailable,
+				Message:            msg,
+				LastTransitionTime: instanceCopy.Status.GetCondition(arcadiav1alpha1.TypeReady).LastTransitionTime,
+				LastSuccessfulTime: metav1.Now(),
```

Comment:
seems most of the code are duplicated, we can use a single object and only use different LastSuccessfulTime if it's the only difference.